### PR TITLE
Fix tests after https://github.com/flare-foundation/flare/pull/23

### PR DESCRIPTION
**Pull request by emailtovamos**
https://github.com/flare-foundation/flare/pull/23 introduced some tests being broken and also there were some tests which were getting skipped. This PR aims to fix all the test:

- [x] windower
- [x] platformvm
- [x] proposervm
- [x] network
- [x] remaining windower
- [x] errPChainHeightNotMonotonic
- [x] Mutate() was deleted but is used in tests. Figure out what to do for this

### CODE REVIEW

**Comment by Ullaakut** [comment]
@emailtovamos Why is this PR a draft at the moment?

**Comment by emailtovamos** [comment]
@Ullaakut I just changed it to ready for review. There are commented out code which I will remove later but the rest should be reviewable.

File: snow/validators/test_retriever.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	GetValidatorsByBlockIDF func(blockID ids.ID) (Set, error)
+}
+
+func (T *TestRetriever) GetValidatorsByBlockID(blockID ids.ID) (Set, error) {
+	if T.GetValidatorsByBlockIDF != nil {
+		return T.GetValidatorsByBlockIDF(blockID)
+	}
+
+	return nil, errGetValidatorsByBlockID
+}
```
Comment (Ullaakut) [inline snow/validators/test_retriever.go:26]: I'd recommend instead just simplifying it to a [simple mock](https://optakt.github.io/code/testing/#mocks) as described in our testing guide.

The mock's func should always be set so that the behavior of the mock is always explicit and clear when reading the test. If there are tons of cases where we need this func call to just not error and it's useless to repeat the same func over and over again, I'd suggest having a mock and a function that builds a BaselineMock as described in the knowledge base as well

```suggestion
type RetrieverMock struct {
	GetValidatorsByBlockIDFunc func(blockID ids.ID) (Set, error)
}

func (m *RetrieverMock) GetValidatorsByBlockID(blockID ids.ID) (Set, error) {
		return m.GetValidatorsByBlockIDF(blockID)
}
```

Also it's very unusual to have an uppercase pointer receiver name 😄

File: snow/validators/test_updater.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	UpdateValidatorsF func(blockID ids.ID) error
+}
+
+func (T *TestUpdater) UpdateValidators(blockID ids.ID) error {
+	if T.UpdateValidatorsF != nil {
+		return T.UpdateValidatorsF(blockID)
+	}
+
+	return errUpdateValidators
+}
```
Comment (Ullaakut) [inline snow/validators/test_updater.go:26]: Same suggestion applies to this test as well

File: vms/platformvm/vm_test.go [hunk 3]
```diff
@@ -524,16 +523,6 @@ func TestGenesis(t *testing.T) {
...
-	// Ensure current validator set of primary network is correct
-	vdrSet, ok := vm.Validators.GetValidators()
-	if !ok {
-		t.Fatalf("Missing the primary network validator set")
-	}
-	currentValidators := vdrSet.List()
-	if len(currentValidators) != 0 {
-		t.Fatal("vm's current validator set is wrong")
-	}
-
```
Comment (Ullaakut) [inline vms/platformvm/vm_test.go:536]: Why is this removed? Currently this wouldn't pass? It seems pretty basic though, that `vm.Validators.GetValidators` should return a non-empty list of validators?

File: vms/proposervm/block.go [hunk 4]
```diff
@@ -82,9 +82,9 @@ func (p *postForkCommonComponents) Verify(parentTimestamp time.Time, parentPChai
 	}
 
 	childPChainHeight := child.PChainHeight()
-	if childPChainHeight < parentPChainHeight {
-		return errPChainHeightNotMonotonic
-	}
+	//if childPChainHeight <= parentPChainHeight { // Since p-chain height doesn't increase for Flare
+	//	return errPChainHeightNotMonotonic
+	//}
```
Comment (Ullaakut) [inline vms/proposervm/block.go:87]: I'd rather remove the code completely than comment it out if that's ok with you?

File: vms/proposervm/post_fork_block_test.go [hunk 5]
```diff
@@ -890,8 +890,8 @@ func TestBlockAccept_PostForkBlock_TwoProBlocksWithSameCoreBlock_OneIsAccepted(t
 	if err != nil {
 		t.Fatal("could not build proBlk2")
 	}
-	if proBlk1.ID() == proBlk2.ID() {
-		t.Fatal("proBlk1 and proBlk2 should be different for this test")
+	if proBlk1.ID() != proBlk2.ID() {
+		t.Fatal("proBlk1 and proBlk2 should not be different for this test")
 	}
```
Comment (Ullaakut) [inline vms/proposervm/post_fork_block_test.go:895]: That's quite an important change, a test was just reversed 😄 How come though? 🤔

File: vms/proposervm/proposer/windower_test.go [hunk 6]
```diff
@@ -18,71 +18,70 @@ import (
...
-	vdrState := &validators.TestState{
-		T: t,
-		GetValidatorSetF: func(height uint64, subnetID ids.ID) (map[ids.ShortID]uint64, error) {
-			return map[ids.ShortID]uint64{
-				validatorID: 10,
-			}, nil
+
+	retriever := &validators.TestRetriever{
+		GetValidatorsByBlockIDF: func(blockID ids.ID) (validators.Set, error) {
+			s := validators.NewSet() //todo use the validatorID in NewSet and NOT use the nonValidatorID
```
Comment (Ullaakut) [inline vms/proposervm/proposer/windower_test.go:48]: Same comment here, TODO should ref an issue or be FIXME

**Comment by Ullaakut** [comment]
@emailtovamos Yeah it seems like some commits from other PRs got mixed up as well. So this PR should be closed in favor of #31 right? If you are confident with the code on #31 you could also just have git pushed with the force argument to replace the code on this PR, which would have avoided having two PRs for it.

**Comment by Ullaakut** [comment]
I'm closing it, feel free to reopen if I misunderstood what you meant.
